From a6777760a2121f02808baecea504ac0e242f860b Mon Sep 17 00:00:00 2001
From: Andrey Konovalov <andrey.konovalov@linaro.org>
Date: Mon, 11 Mar 2024 15:15:08 +0100
Subject: [PATCH 04/21] libcamera: internal: Move SharedMemObject class to a
 common directory

Move SharedMemObject class out of RPi namespace and put it into
include/libcamera/internal so that everyone could use it.

Tested-by: Bryan O'Donoghue <bryan.odonoghue@linaro.org> # sc8280xp Lenovo x13s
Tested-by: Pavel Machek <pavel@ucw.cz>
Reviewed-by: Pavel Machek <pavel@ucw.cz>
Signed-off-by: Andrey Konovalov <andrey.konovalov@linaro.org>
Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Reviewed-by: Milan Zamazal <mzamazal@redhat.com>
---
 include/libcamera/internal/meson.build                      | 1 +
 .../libcamera/internal}/shared_mem_object.h                 | 6 +-----
 2 files changed, 2 insertions(+), 5 deletions(-)
 rename {src/libcamera/pipeline/rpi/common => include/libcamera/internal}/shared_mem_object.h (97%)

diff --git a/include/libcamera/internal/meson.build b/include/libcamera/internal/meson.build
index 33eb0fb3..5807dfd9 100644
--- a/include/libcamera/internal/meson.build
+++ b/include/libcamera/internal/meson.build
@@ -39,6 +39,7 @@ libcamera_internal_headers = files([
     'process.h',
     'pub_key.h',
     'request.h',
+    'shared_mem_object.h',
     'source_paths.h',
     'sysfs.h',
     'v4l2_device.h',
diff --git a/src/libcamera/pipeline/rpi/common/shared_mem_object.h b/include/libcamera/internal/shared_mem_object.h
similarity index 97%
rename from src/libcamera/pipeline/rpi/common/shared_mem_object.h
rename to include/libcamera/internal/shared_mem_object.h
index aa56c220..98636b44 100644
--- a/src/libcamera/pipeline/rpi/common/shared_mem_object.h
+++ b/include/libcamera/internal/shared_mem_object.h
@@ -6,8 +6,8 @@
  */
 #pragma once
 
-#include <cstddef>
 #include <fcntl.h>
+#include <stddef.h>
 #include <string>
 #include <sys/mman.h>
 #include <sys/stat.h>
@@ -19,8 +19,6 @@
 
 namespace libcamera {
 
-namespace RPi {
-
 template<class T>
 class SharedMemObject
 {
@@ -123,6 +121,4 @@ private:
 	T *obj_;
 };
 
-} /* namespace RPi */
-
 } /* namespace libcamera */
-- 
2.43.2

